File: Data Structures & Algorithms/maximum-subarray/submission-0.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
            int  n = nums.size();
            if(n<=0){
                return 0;
            }
            vector<int>dp(n+1,0);
            dp[0]=nums[0];
            int maxi=nums[0];
            for(int i=1;i<n;i++){
                dp[i]= nums[i]+ max(0,dp[i-1]);
                maxi=max(maxi,dp[i]);
            }
            return maxi;
    }
};

File: Data Structures & Algorithms/max-consecutive-ones/submission-0.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        seen1 = False
        maxctr = 0
        curctr=0
        for num in nums:
            if num == 1:
                if not seen1:
                    seen1= True
                    
                curctr+=1
            if num ==0:
                if seen1:
                    maxctr= max(maxctr,curctr)
                    curctr= 0
                    seen1 = False
        return max(maxctr,curctr)

                